internal/pkg/secret/encryption_other.go

```diff
@@ -0,0 +1,76 @@
+//go:build linux || darwin
+
+package secret
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/sha256"
+	"encoding/base64"
+	"os"
+	"strconv"
+
+	"github.com/denisbrodbeck/machineid"
+
+	"golang.org/x/crypto/pbkdf2"
+)
+
+func DeriveEncryptionKey(salt string) ([]byte, error) {
+	machineId, err := machineid.ID()
+	if err != nil {
+		return []byte{}, err
+	}
+	userId := strconv.Itoa(os.Getuid())
+	encryptionKey := pbkdf2.Key([]byte(machineId+userId), []byte(salt), 10000, 32, sha256.New)
+	return encryptionKey, nil
+}
+
+func Encrypt(password, salt string) (string, error) {
+	encryptionKey, err := DeriveEncryptionKey(salt)
+	if err != nil {
+		return "", err
+	}
+
+	block, err := aes.NewCipher(encryptionKey)
+	if err != nil {
+		return "", err
+	}
+
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return "", err
+	}
+
+	encryptedPassword := aesgcm.Seal(nil, encryptionKey[:12], []byte(password), nil)
```

Comment:
Last parameter of this Seal method is AAD (authenticated addition data) that has special meaning in AES GCM. It provides authenticated encryption that ensures integrity along with confidentiality. I would suggest to use logged in user's email as AAD. We are anyway going to re-encrypt the password if user login next time in CLI.

Reply:
Password decryption has to work when the user is not logged in and no username (email) is provided. I suppose the same aad will need to be provided to `aesgcm.Open` as well?

Reply:
Yes same AAD has to be provided while decryption. 

> Password decryption has to work when the user is not logged in and no username (email) is provided.
When will this happen? I believe we maintain user's email id in config somewhere and that can be directly used. We do not need to ask user to provide email that can be used in AAD.